src/EnduranceTests/Targets/SlackTarget.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace NServiceBus.AzureServiceBus.EnduranceTests.Targets
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Specialized;
+    using System.Net;
+    using NLog;
+    using NLog.Targets;
+    using Newtonsoft.Json;
+    using NLog.Config;
+    using NLog.Layouts;
+    using NServiceBus.AzureServiceBus.EnduranceTests.TestUtils;
+
+    [Target("Slack")]
+    public sealed class SlackTarget : TargetWithLayout
+    {
+        private Uri _uri = null;
+
+        [RequiredParameter]
+        public string WebhookUri { get { return _uri?.ToString() ?? string.Empty; } set { _uri = new Uri(value);} }
+
+        public string StackTraceLayout { get; set; } = "${stacktrace}";
+
+        protected override void Write(LogEventInfo logEvent)
+        {
+            var logMessage = Layout.Render(logEvent);
+
+            SendMessageToSlack(new SlackMessage
+            {
+                attachments = new List<SlackAttachment>
+                {
+                    new SlackAttachment
+                    {
+                        title = logEvent.LoggerName,
+                        title_link = BlobNameBuilder.Build(TestEnvironment.AzureStorage, logEvent),
+                        text = logMessage,
+                        fallback = logMessage,
+                        color = "danger"
+                    },
+                    new SlackAttachment
+                    {
+                        text = (new SimpleLayout(StackTraceLayout)).Render(logEvent)
+                    }
+                }
+            });
+        }
+
+        private void SendMessageToSlack(SlackMessage message)
+        {
+            using (var client = new WebClient())
+            {
+                var data = new NameValueCollection()
+                {
+                    {"payload", JsonConvert.SerializeObject(message)}
+                };
+
+                client.UploadValuesAsync(_uri, "POST", data);
```

Comment:
Async method call in a void message? 
If can't async all the way, could do `client.UploadValuesAsync(_uri, "POST", data).GetAwaiter().GetResult